calcom/cal.com
# feat: date range overhaul

Would like 3 approvals before merge 🙏 

## What does this PR do?

Rewrites slot/availability logic by using date ranges. It simplifies the code, it was really tricky to make fixes in the old logic. It also fixes the issue that date overrides in different timezones didn't adjust, resulting in the organizer being booked outside their availability. All existing tests are passing + added new tests for the new logic

### How should it be tested?
  - check if all test pass with `yarn test`
  - check if the available slots shown for an event are correct with different combinations of event type settings and time zones: 
    -  organizer and attendee timezones 
    - working hours
    - date overrides 
    - event type setting (min booking notice, buffers, event duration, etc.)
     	

Fixes that date overrides were not adjusted to different timezones:
Fixes #8666
Fixes #8374
Fixes #8329
Fixes #8000
Fixes #7687
Fixes #7114
